alf/environments/suite_dmc.py

```diff
@@ -11,121 +11,79 @@
     For installation of MuJoCo210, see https://mujoco.org.
 
     Args:
-        environment_name (str): Do not use this arg, this arg is here to
-            match up with create_environment.
-        domain_name (str): The name of MuJoCo domain that is used.
-        task_name (str): The name of task we want the agent to do in the
-            current MuJoCo domain.
-        seed (int): Random seed for the environment. This parameter is
-            here to serve as a place holder and will be override by the
-            seed of create_environment().
+        environment_name (str): this string must have the format
+            "domain_name:task_name", where "domain_name" is defined by DM control as
+            the physical model name, and "task_name" is an instance of the model
+            with a parcular MDP structure.
         from_pixels (boolean): Output image if set to True.
-        image_size (int): The height and width of the output 
+        image_size (int): The height and width of the output
             image from the environment.
-        action_repeat (int): Action repeat of gym environment.
         env_id (int): (optional) ID of the environment.
         discount (float): Discount to use for the environment.
+        visualize_reward: if True, then the rendered frame will have
+            a highlighted color when the agent achieves a reward.
         max_episode_steps (int): The maximum episode step in the environment.
             Note that the episode length in the alf will be max_episode_steps/action_repeat.
         gym_env_wrappers (Iterable): Iterable with references to gym_wrappers
             classes to use directly on the gym environment.
         alf_env_wrappers (Iterable): Iterable with references to alf_wrappers
-            classes to use on the ALF environment. There will be an 
+            classes to use on the ALF environment. There will be an
             AlfEnvironmentDMC2GYMWrapper added before any alf_wrappers.
-        
+
     Returns:
         A wrapped AlfEnvironment
     """
+    names = environment_name.split(":")
+    assert len(names) == 2, (
+        "environment_name must be in the format 'domain_name:task_name'!"
+        f" Provided environment_name: {environment_name}")
 
-    gym_env = dmc2gym.make(
+    domain_name, task_name = names
+    gym_env = DMCGYMWrapper(
         domain_name=domain_name,
         task_name=task_name,
-        seed=seed,
-        visualize_reward=False,
+        visualize_reward=visualize_reward,
         from_pixels=from_pixels,
-        # The episode_length passed to dmc2gym is doubled because we want to
+        # The episode_length passed to DMCGYMWrapper is doubled because we want to
         # make sure that the reset is triggered only in alf_wrappers.TimeLimit.
         episode_length=max_episode_steps * 2,
         height=image_size,
-        width=image_size,
-        frame_skip=action_repeat)
```

Comment:
episode_length may not be large enough if using FrameSkip wrapper to replace this. episode_length needs to be bigger than max_episode_steps * action_repeat

Reply:
Now always pass into episode_length=float('inf')